### PR TITLE
Update to latest signalflow client

### PR DESCRIPTION
**Pull request by keitwb**
Change some time related types to native types for more intuitive manipulation

### CODE REVIEW

File: pkg/job_handler.go [hunk 1]
```diff
@@ -111,58 +106,51 @@ func (t *SignalFxJobHandler) reuse(target *Target) (chan []*datasource.TimeSerie
 
 func (t *SignalFxJobHandler) isJobReusable(target *Target) bool {
 	return t.program == target.Program &&
-		t.intervalMs == target.IntervalMs &&
-		t.startTime <= target.StartTime &&
-		((t.channel != nil && t.unbounded) || t.stopTime >= target.StopTime)
+		t.interval == target.Interval &&
+		t.startTime.Before(target.StartTime) &&
```
Comment (piotr-signalfx) [inline pkg/job_handler.go]: !t.startTime.After(target.StartTime)

File: pkg/job_handler.go [hunk 2]
```diff
@@ -111,58 +106,51 @@ func (t *SignalFxJobHandler) reuse(target *Target) (chan []*datasource.TimeSerie
...
+		((t.computation != nil && t.unbounded) || t.stopTime.After(target.StopTime))
```
Comment (piotr-signalfx) [inline pkg/job_handler.go]: !t.stopTime.Before(target.StopTime)

File: pkg/job_handler.go (hunk 1 above)
Comment (piotr-signalfx) [inline pkg/job_handler.go]: !t.startTime.After(target.StartTime)

File: pkg/job_handler.go [hunk 3]
```diff
@@ -111,58 +106,51 @@ func (t *SignalFxJobHandler) reuse(target *Target) (chan []*datasource.TimeSerie
...
 }
 
-func (t *SignalFxJobHandler) readMessages() {
-	for msg := range t.channel.Messages() {
-		if dm, ok := msg.(*signalflow.DataMessage); ok {
+func (t *SignalFxJobHandler) readDataMessages() {
+	for {
+		select {
+		// This channel receives when there is no more data
+		case <-t.computation.Done():
```
Comment (piotr-signalfx) [inline pkg/job_handler.go]: What will happen on error? Will also a message be received through this channel?

File: pkg/job_handler.go [hunk 4]
```diff
@@ -111,58 +106,51 @@ func (t *SignalFxJobHandler) reuse(target *Target) (chan []*datasource.TimeSerie
...
 	}
-	if t.resolution > 0 {
+	resolution := t.computation.Resolution()
+	if resolution > 0 {
+		maxDelay := t.computation.MaxDelay()
 		// Estimate the timestamp of the last datapoint already available in the system
-		nextEstimatedTimestamp := timestamp + int64(math.Ceil(float64(t.maxDelay)/float64(t.resolution)+1))*t.resolution
-		roundedCutoffTime := int64(math.Floor(float64(t.cutoffTime)/float64(t.resolution))) * t.resolution
-		return nextEstimatedTimestamp > roundedCutoffTime
+		nextEstimatedTimestamp := timestamp.Add(resolution + maxDelay)
```
Comment (piotr-signalfx) [inline pkg/job_handler.go]: nextEstimatedTimestamp should be always aligned to the next interval hence the need to round up maxDelay. Something like:
`nextEstimatedTimestamp := timestamp.Add(resolution + maxDelay.Add(resolution - 1).Truncate(resolution))`

File: pkg/job_handler.go [hunk 5]
```diff
@@ -180,42 +168,14 @@ func toFloat64(value interface{}) float64 {
...
-		n, _ := strconv.ParseInt(fmt.Sprintf("%v", m.MessageBlock.Contents["maxDelayMs"]), 10, 64)
-		t.maxDelay = n
-		t.logger.Debug("Max Delay", "value", t.resolution)
-	default:
-		log.Printf("Message: %v", m.MessageBlock)
-	}
-}
-
-func (t *SignalFxJobHandler) handleMetadataMessage(m *signalflow.MetadataMessage) {
+func (t *SignalFxJobHandler) handleMetadataMessage(m *messages.MetadataMessage) {
```
Comment (piotr-signalfx) [inline pkg/job_handler.go]: Can be removed

File: .gitignore [hunk 6]
```diff
@@ -15,4 +15,6 @@ package-lock.json
 yarn.lock
 Gopkg.lock
 
-/vendor
\ No newline at end of file
+dist/
```
Comment (piotr-signalfx) [inline .gitignore]: We cannot ignore dist, see https://github.com/signalfx/grafana-signalfx-datasource#installation :)

File: Makefile [hunk 7]
```diff
@@ -1,12 +1,14 @@
 all: grunt deps build
```
Comment (piotr-signalfx) [inline Makefile]: Running make results in:
```
dep ensure
could not find project Gopkg.toml, use dep init to initiate a manifest
make: *** [deps] Error 1
```

File: pkg/job_handler.go (hunk 3 above)
Comment (keitwb) [reply pkg/job_handler.go]: Errors in the computation (e.g. websocket connect dropped, or a CHANNEL_ABORT or END_OF_CHANNEL message, etc.) will also cause this to trigger and cancel this job.

I need to work out a good interface to save and get errors about the client as well as attempt some retry functionality on well-known errors like a websocket connection drop.

File: pkg/job_handler.go (hunk 4 above)
Comment (keitwb) [reply pkg/job_handler.go]: Used a slightly different way of writing it but I think it is equivalent.

**Pull request merged**